sync/src/types/mod.rs

```diff
@@ -69,13 +69,60 @@ const SHRINK_THRESHOLD: usize = 300;
 //     If their best known block is still behind when that new timeout is
 //     reached, disconnect.
 
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug)]
```

Comment:
You can derive `Default` for `HeadersSyncState` so `ChainSyncState` can continue to use auto derive.